Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -189,7 +188,7 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
                           selectionMethod: NetworkProtectionServerSelectionMethod) async throws -> (server: NetworkProtectionServer,
                                                                                                     newExpiration: Date?) {
 
-        guard let token = try? tokenStore.fetchToken() else { throw NetworkProtectionError.noAuthTokenFound }
+        let token = try await VPNAuthTokenBuilder.getVPNAuthToken(from: tokenProvider, policy: .localValid)
```

Comment:
Could you wrap this in a `NetworkProtectionError` with underlying error data?

It'll make it much easier to debug issues if we know where this error is coming from.

Reply:
I have modified Networking.OAuthClientError.missingTokens by adding an underlying error and used in here